lib/ui/platform_isolate.dart

```diff
@@ -0,0 +1,169 @@
+      // pause and terminate capabilities aren't provided to the parent
+      // isolate, and errors are fatal is false. But if the isolate does
+      // shutdown unexpectedly, clear the singleton so we can create another.
+      for (final Completer<Object?> completer in _pending.values) {
+        completer.completeError(RemoteError('PlatformIsolate shutdown unexpectedly', StackTrace.empty.toString()));
+      }
+      _pending.clear();
+      _platformRunnerSendPort = null;
+      _platformRunnerSendPortFuture = null;
+    } else if (message is _PlatformIsolateReadyMessage) {
+      _platformRunnerSendPort = message.computationPort;
+      sendPortCompleter.complete(message.computationPort);
+    } else if (message is _ComputationResult) {
+      final Completer<Object?> resultCompleter = _pending.remove(message.id)!;
+      final Object? remoteStack = message.remoteStack;
+      final Object? remoteError = message.remoteError;
+      if (remoteStack != null) {
+        if (remoteStack is StackTrace) {
+          // Typed error.
+          resultCompleter.completeError(remoteError!, remoteStack);
+        } else {
+          // onError handler message, uncaught async error.
+          // Both values are strings, so calling `toString` is efficient.
+          final RemoteError error =
+              RemoteError(remoteError!.toString(), remoteStack.toString());
+          resultCompleter.completeError(error, error.stackTrace);
+        }
+      } else {
+        resultCompleter.complete(message.result);
+      }
+    } else {
+      // We encountered an error while starting the new isolate.
+      throw IsolateSpawnException('Unable to spawn isolate: $message');
+    }
+  };
+  final Isolate parentIsolate = Isolate.current;
+  final SendPort sendPort = receiver.sendPort;
+  try {
+    _nativeSpawn(() => _platformIsolateMain(parentIsolate, sendPort));
+  } on Object {
+    receiver.close();
+    rethrow;
+  }
+  return sendPortCompleter.future;
+}
+
+Future<R> _sendComputation<R>(
+    SendPort port, FutureOr<R> Function() computation) {
+  final int id = ++_nextId;
+  final Completer<R> resultCompleter = Completer<R>();
+  _pending[id] = resultCompleter;
+  port.send(_ComputationRequest(id, computation));
+  return resultCompleter.future;
+}
+
+void _platformIsolateMain(Isolate parentIsolate, SendPort sendPort) {
+  final RawReceivePort computationPort = RawReceivePort();
+  computationPort.handler = (_ComputationRequest? message) {
+    if (message == null) {
+      // The parent isolate has shutdown. Allow this isolate to shutdown.
+      computationPort.keepIsolateAlive = false;
+      return;
+    }
+    try {
+      final FutureOr<Object?> potentiallyAsyncResult = message.computation();
+      if (potentiallyAsyncResult is Future<Object?>) {
+        potentiallyAsyncResult.then((Object? result) {
+          sendPort.send(_ComputationResult(message.id, result, null, null));
```

Comment:
I believe if this `sendPort.send()` throws the error is uncaught in platform isolate and receiver never gets an answer. It needs a try/catch around I believe.

Consider adding a test that returns an object that cannot be sent across ports.

Consider also structuring the code differently (to e.g. not think the outermost try/catch will catch the error here):

```dart
late final FutureOr<Object?> potentiallyAsyncResult;
try {
  potentiallyAsyncResult = message.computation();
} catch (error, stack) {
  sendPort.send(_ComputationResult(message.id, null, error, stack));
  return;
}

// Sync result
if (potentiallyAsyncResult is! Future) {
  try {
    sendPort.send(_ComputationResult(message.id, null, error, stack));
  } ... { ... }
  return;
}

// Async result
potentiallyAsyncResult.then((result) {
 try { ... } catch { ... }
}, onError: (error, stack) {
 try { ... } catch { ... }
});

```